File: python/playwright-sync-captcha/scrape.py
#!/usr/bin/env python3
from os import environ
from playwright.sync_api import Playwright, sync_playwright

AUTH = environ.get('AUTH', default='USER:PASS')
TARGET_URL = environ.get('TARGET_URL', default='https://example.com')


def scrape(playwright: Playwright, url=TARGET_URL):
    if AUTH == 'USER:PASS':
        raise Exception('Provide Scraping Browsers credentials in AUTH ' +
                        'environment variable or update the script.')
    print('Connecting to Browser...')
    endpoint_url = f'wss://{AUTH}@brd.superproxy.io:9222'
    browser = playwright.chromium.connect_over_cdp(endpoint_url)
    try:
        print(f'Connected! Navigating to {url}...')
        page = browser.new_page()
        client = page.context.new_cdp_session(page)
        page.goto(url, timeout=2*60_000)
        print('Navigated! Waiting captcha to detect and solve...')
        result = client.send('Captcha.waitForSolve', {
            'detectTimeout': 10 * 1000,
        })
        status = result['status']
        print(f'Captcha status: {status}')
    finally:
        browser.close()


def main():
    with sync_playwright() as playwright:
        scrape(playwright)


if __name__ == '__main__':
    main()

File: golang/playwright-captcha/scrape.go
package main

import (
	"errors"
	"fmt"
	"log"
	"os"

	"github.com/playwright-community/playwright-go"
)

func connect() (playwright.Browser, error) {
	err := playwright.Install(&playwright.RunOptions{
		SkipInstallBrowsers: true,
	})
	if err != nil {
		return nil, err
	}
	pw, err := playwright.Run()
	if err != nil {
		return nil, err
	}
	auth, ok := os.LookupEnv("AUTH")
	if !ok {
		return nil, errors.New("Provide Scraping Browsers credentials in AUTH environment variable or update the script.")
	}
	endpointURL := fmt.Sprintf("wss://%s@brd.superproxy.io:9222", auth)
	return pw.Chromium.ConnectOverCDP(endpointURL)
}

func scrape(url string) (string, error) {
	fmt.Print("Connecting to Browser...\n")
	br, err := connect()
	if err != nil {
		return "", err
	}
	defer br.Close()
	fmt.Printf("Connected! Navigating to %s...\n", url)
	page, err := br.NewPage()
	if err != nil {
		return "", err
	}
	session, err := page.Context().NewCDPSession(page)
	if err != nil {
		return "", err
	}
	_, err = page.Goto(url)
	if err != nil {
		return "", err
	}
	fmt.Print("Navigated! Waiting captcha to detect and solve...\n")
	result, err := session.Send("Captcha.waitForSolve", map[string]any{
		"detectTimeout": 30000,
	})
	if err != nil {
		return "", err
	}
	status, _ := result.(map[string]interface{})["status"].(string)
	fmt.Printf("Captcha status: %s\n", status)
	return status, nil
}

func main() {
	url, ok := os.LookupEnv("TARGET_URL")
	if !ok {
		url = "https://example.com"
	}
	_, err := scrape(url)
	if err != nil {
		log.Fatal(err)
	}
}
